File: src/backend/PROJECTTypes.py
from typing import NamedTuple

class TemperatureReading(NamedTuple):
    current: float | None
    high: float | None
    critical: float | None

class UsersSession(NamedTuple):
    name: str | None
    terminal: str | None
    host: str | None
    started: float | None
    pid: int | None

File: src/frontend/src/types/projectTypes.ts
export interface TemperatureReading { 
    [index: number]: { current: number; high: number; critical: number; }
}

export interface UserSession {
    name: string | null,
    terminal: string | null,
    host: string | null,
    started: number | null,
    pid: number | null
}

export interface LoadReading {
    type: string | null,
    value: number | null
}